### ISSUE TITLE
Templates list: info on whether a template has been customized should be accessible

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#42557]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/issues/42505

In the Templates list, the blue dot to indicate a template has been customized shows some info only on hover. There's no other indication available that informs keyboard users or assistive technology users a template has been customized.

There should be some text to provide this information in an accessible way. 

I'd say the blue dot isn't very clear also for sighted users. It took some time to me to understand what it was about. And I had to hover the icon to get the related information. Visible text would benefit all users.

Also, this is a _data_ table. I'd argue that the column cells should only contain the data type identified by the column header. The 'Added by' column should only contain info on who/what added the template. Instead, the text 'This template has been customized' is about the template _state_. It's a data that doesn't fit at all with 'Added by' and ideally should go in a separate column dedicated to the template state.

### Step-by-step reproduction instructions

- Make some changes to a default template and save.
- Click Toggle navigation, then click Templates to open the Templates list.
- Observe the 'Added by' column of the template you customized.
- Observe the template icon shows a blue dot.
- Hover the icon and observe a tooltip appears with text 'This template has been customized'.
- Stop hovering the icon.
- Observe the 'This template has been customized' text is not present in any other place on the page (not even in some hidden text in the source).
- This important information is not available to keyboard users and screen reader users.

### Screenshots, screen recording, code snippet

<img width="882" alt="template customized" src="https://user-images.githubusercontent.com/1682452/179962757-277e980f-eab4-41fa-a728-fafbb7c3fbf7.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
@jameskoster @javierarce thanks for your feedback on https://github.com/WordPress/gutenberg/issues/42505, https://github.com/WordPress/gutenberg/pull/42555, https://github.com/WordPress/gutenberg/pull/42589, and https://github.com/WordPress/gutenberg/pull/47841.

To recapt:
- A text-based solution was proposed on https://github.com/WordPress/gutenberg/pull/42589.
- Splitting the table, as suggested in https://github.com/WordPress/gutenberg/pull/42555#issuecomment-1191580202, was explored in https://github.com/WordPress/gutenberg/pull/47841

It appears we all feel the templates/parts list needs improvements under various aspects that should be explored also under the light of future plans and new features of the templates list. For now, I suggest to focus on the specific problem of this issue: the information about the 'customized' state.

This:

<img width="306" alt="Screenshot 2023-02-13 at 15 08 01" src="https://user-images.githubusercontent.com/1682452/218481977-dafab372-d044-483c-87e4-c35e921bf355.png">

ins't that great, as the information is only visual and the tooltip appears only on hover. It's not accessible.

Also, when the list of templates/parts is very long, it's very hard to immediately identify the customized templates/parts:

<img width="1442" alt="Screenshot 2023-02-13 at 12 04 28" src="https://user-images.githubusercontent.com/1682452/218482827-9ba9b30d-8219-4ec6-bf60-e59fda001a3c.png">

It's a relatively simple issue and it's waiting for a solution since more than 6 months. I'd like to suggest the simplest option: just add a second line in the 'Added by' cell with text: "User customized", as suggested in https://github.com/WordPress/gutenberg/pull/42589#issuecomment-1192220187

**Comment by jameskoster** [comment]
I agree, it would be good to implement a light-weight solution that solves the issue now, and doesn't paint us into a corner with regards to future enhancements.

The suggestion can work, and we might even consider removing the blue dot. With the accompanying text it's not really that useful:

<img width="562" alt="Screenshot 2023-02-13 at 16 15 52" src="https://user-images.githubusercontent.com/846565/218512138-7a74def6-f4d3-4fbf-a56f-2bc78ed49fce.png">



The associated user is also removed. I'm not sure how useful that information is.

> Also, when the list of templates/parts is very long, it's very hard to immediately identify the customized templates/parts:

Imo this part would be best served by filtering. Indeed we can use filtering to address a number of similar flows that are currently quite painful e.g:

* View all templates added by a specific theme or plugin
* View all the 'blog' templates
* View templates created by a specific author
